agent/internal/mapper/grpc.go

```diff
@@ -70,42 +71,180 @@ func MapDeployImage(req *agent.DeployRequest, appConfig *config.CommonConfigurat
 	return res
 }
 
-func MapContainerConfig(in *agent.DeployRequest_ContainerConfig) v1.ContainerConfig {
+func MapContainerConfig(in *agent.DeployRequest) v1.ContainerConfig {
 	containerConfig := v1.ContainerConfig{
-		Container: in.Name,
-		Ports:     MapPorts(in.Ports),
-		User:      &in.User,
+		Container:        in.Name,
+		ContainerPreName: in.InstanceConfig.Prefix,
+		Ports:            MapPorts(in.ContainerConfig.Ports),
+		PortRanges:       MapPortRanges(in.ContainerConfig.PortRanges),
+		Volumes:          MapVolumes(in.ContainerConfig.Volumes),
+		User:             in.ContainerConfig.User,
 	}
 
-	if in.Prefix != nil {
-		containerConfig.ContainerPreName = *in.Prefix
+	if in.ContainerConfig.TTY != nil {
```

Comment:
i think we could just lift in.ContainerConfig to a variable cc and use that instead of in.ContainerConfig